### PR TITLE
Update chunk format docs with native histograms and OOO

### PR DESCRIPTION
**Pull request by fionaliao**
Part of https://github.com/prometheus/prometheus/issues/12005. Updating format/chunks.md and format/head_chunks.md with float native histograms and out of order changes.

### CODE REVIEW

File: tsdb/docs/format/chunks.md [hunk 1]
```diff
@@ -107,9 +107,9 @@ Notes:
...
-└────────────────────────┴───────────────────────────┴────────────────────────────────┴──────────────────────┴─────────────────────────────────┴─────┴─────────────────────────────────┴─────────────────────────────────┴─────┴─────────────────────────────────┘
+┌───────────────────────┬──────────────────────────┬───────────────────────────────┬──────────────────────┬─────────────────────────────────┬─────┬─────────────────────────────────┬─────────────────────────────────┬─────┬─────────────────────────────────┐
+│ ts_delta <varbit_int> │ count_delta <varbit_int> │ zero_count_delta <varbit_int> │ sum_xor <varbit_xor> │ pos_bucket_0_delta <varbit_int> │ ... │ pos_bucket_n_delta <varbit_int> │ neg_bucket_0_delta <varbit_int> │ ... │ neg_bucket_n_delta <varbit_int> │
+└───────────────────────┴──────────────────────────┴───────────────────────────────┴──────────────────────┴─────────────────────────────────┴─────┴─────────────────────────────────┴─────────────────────────────────┴─────┴─────────────────────────────────┘
```
Comment (fionaliao) [inline tsdb/docs/format/chunks.md:112]: This was corrected from varbit_uint to varbit_int, see: https://github.com/prometheus/prometheus/blob/105ab2e95afa786c7ad21be614a31ff45c143cbb/tsdb/chunkenc/histogram.go#L656-L658

**Review by bboreham — APPROVED** [review]
I don't know the tiny details but seems legit.

**Comment by beorn7** [comment]
Thank you very much. Will have a look ASAP.

**Review by jesusvazquez — APPROVED** [review]
Looking good to me!

**Review by Tusenka — APPROVED** [review]

File: tsdb/docs/format/chunks.md [hunk 2]
```diff
@@ -36,7 +36,7 @@ in-file offset (lower 4 bytes) and segment sequence number (upper 4 bytes).
 
 Notes:
 * `<uvarint>` has 1 to 10 bytes.
-* `encoding`: Currently either `XOR` or `histogram`.
+* `encoding`: Currently either `XOR`, `histogram`, or `float histogram`.
```
Comment (beorn7) [inline tsdb/docs/format/chunks.md:39]: `float histogram` → `floathistogram` (to match the string returns of `Encoding.String()` in chunkenc/chunk.go).

**Review by beorn7 — APPROVED** [review]
Looks great, thank you very much.

What's still missing is NHCB. Maybe it's only the _Histogram chunk data_ section that needs an update. In any case, we can do that in a separate PR.

**Comment by beorn7** [comment]
I'm already working on the NHCB PR.

**Comment by beorn7** [comment]
https://github.com/prometheus/prometheus/pull/15135

File: tsdb/docs/format/chunks.md (hunk 2 above)
Comment (beorn7) [reply tsdb/docs/format/chunks.md:39]: I forgot I made this comment when merging. I'll include the change in https://github.com/prometheus/prometheus/pull/15135 .

**Pull request merged**